resource-synchronizer/poet_ready.py

```diff
@@ -0,0 +1,88 @@
+import os
+import json
+import shutil
+import sys
+
+RB_DIR = 'rollback'
+CODE_DIR = os.environ.get('CODE_DIR')
+
+
+def cleanup():
+
+    shutil.rmtree(RB_DIR)
+
+
+def rollback():
+    CODE_DIR = os.environ.get('CODE_DIR')
+    try:
+        shutil.rmtree('.vscode')
+    except:
+        pass
+
+    try:
+        os.remove('.gitignore')
+    except:
+        pass
+
+    for root, dirs, files in os.walk(RB_DIR):
+
+        for d in dirs:
+            path_dir = os.path.join(root, d)
+            dest_path_dir = path_dir.replace(RB_DIR, CODE_DIR)
+            os.mkdir(dest_path_dir)
+
+        for f in files:
+            path_file = os.path.join(root, f)
+            dest_path_file = path_file.replace(RB_DIR, CODE_DIR)
+            shutil.copy2(path_file, dest_path_file)
+
+    cleanup()
+
+
+def main():
+    os.mkdir(RB_DIR)
+    CODE_DIR = os.environ.get('CODE_DIR')
+    try:
+        poet_files = f'{CODE_DIR}/poet.json'
+        with open(poet_files, 'r') as pf:
+            poet_files = json.loads(pf.read())
+
+        # handling .gitignore
+        try:
+            shutil.move('.gitignore', RB_DIR)
+        except FileNotFoundError:
+            pass
+
+        gitignore_path = os.path.join(CODE_DIR, '.gitignore')
+        with open(gitignore_path, 'x') as f:
+            f.write('\n'.join(poet_files['gitignore']))
+
+        # handling .vscode/settings.json
+        dest_dir = os.path.join(CODE_DIR, '.vscode')
+        try:
+            shutil.move(dest_dir, RB_DIR)
+        except FileNotFoundError:
+            pass
+
+        os.mkdir(dest_dir)
+        file = '/settings.json'
+        dest_path = dest_dir + file
+
+        with open(dest_path, 'x') as f:
+            json.dump(poet_files['vscode'], f, indent=2)
+
+        # handling readme
+        try:
+            read_me = os.path.join(CODE_DIR, 'README.md')
+            shutil.move(read_me, RB_DIR)
+        except FileNotFoundError:
+            pass
+
+        cleanup()
+    except:
+        rollback()
+        sys.exit()
```

Comment:
It looks like we're trying to make this script handle errors gracefully so that it never fails. However, in practice could that cause a pipeline to succeed (since the script won't error and rollback to the original files) when things haven't been updated as expected? If so, would it be preferable to let it fail hard?

Reply:
Ah I need to update that to `sys.exit(1)` the intention is for the except block to still fail hard, but rolling back any files that were created and setting the repo back to it's original state in regards to `.vscode/settings.json` and `.gitignore`.